Functional task

anna-kapitonova opened the pull request.

anna-kapitonova committed:
Create README.md

anna-kapitonova committed:
added script

anna-kapitonova committed:
Update README.md

anna-kapitonova committed:
Delete pain.py

anna-kapitonova committed:
Update functional.py

krglkvrmn commented on HW6_functional/functional.py:
```diff
@@ -0,0 +1,43 @@
+
+
+def conditional_reduce(func1, func2, cont):
+    good_cont = list()
+    for i in cont:
+        if func1(i):
+            good_cont.append(i)
```
Можно и так
```suggestion
    good_cont = list(filter(func1, cont))
```

krglkvrmn commented on HW6_functional/functional.py:
```diff
@@ -0,0 +1,43 @@
+    for i in cont:
+        if func1(i):
+            good_cont.append(i)
+    return func2(good_cont[0], good_cont[1])
```
`reduce` должен работать с последовательностью длины 2 и более. Хорошо про это написано в документации 
![image](https://user-images.githubusercontent.com/48160953/145684396-dd18aec4-aa1a-4f47-9c7a-3638bf4ca40b.png)

krglkvrmn commented on HW6_functional/functional.py:
```diff
@@ -0,0 +1,43 @@
+def sequential_map(*funcs):
```
Имелось ввиду, что функции последовательно применяются к **каждому** занчению в контейнере, а не ко всему сразу

krglkvrmn commented on HW6_functional/functional.py:
```diff
@@ -0,0 +1,43 @@
+    return res
+
+
+def sequential_map_new(*funcs):
```
Аналогично как и в случае с `sequential_map`, `func_chain` должна работать с 1 значением, а не со всем контейнером

krglkvrmn reviewed (commented):
Хорошая работа, но, видимо, нужно еще немного разобраться с `reduce`. С отстальными заданиями скорее всего просто недопонимание вышло, так что много не снимал. Комментарии ниже

anna-kapitonova committed:
Roma's suggested change of good_cont

Co-authored-by: Roman Kruglikov <48160953+krglkvrmn@users.noreply.github.com>

anna-kapitonova committed:
updated readme

anna-kapitonova committed:
edited functions

now they work even if lambda is the first function and conditional_reduce works as it should

anna-kapitonova committed:
Merge branch 'main' into functional
